OpenRA.Game/Sound/Sound.cs

```diff
@@ -348,9 +350,20 @@ public float VideoSeekPosition
 			var name = prefix + clip + suffix;
 
 			if (!string.IsNullOrEmpty(name) && (p == null || p == p.World.LocalPlayer))
-				soundEngine.Play2D(sounds[name],
+			{
+				var sound = soundEngine.Play2D(sounds[name],
 					false, relative, pos,
 					InternalSoundVolume * volumeModifier, attenuateVolume);
+				if (id != 0)
+				{
+					if (currentSounds.ContainsKey(id))
+						soundEngine.StopSound(currentSounds[id]);
+
+					currentSounds.Remove(id);
```

Comment:
Isn't this redundant (since the following line would overwrite this entry anyway)?